src/pages/VehicleRegister/VehicleRegister.test.jsx

```diff
@@ -0,0 +1,253 @@
+        const btn = screen.getByRole("button", { name: /cadastrar/i });
+        userEvent.type(model, mockVehicle.model);
+        userEvent.type(year, mockVehicle.year);
+        userEvent.type(price, mockVehicle.price);
+        userEvent.selectOptions(brand, mockVehicle.brand.nome);
+        await act(async () => userEvent.click(btn));
+      });
+
+      describe("And the response is successful", () => {
+        it("should register vehicle with correct data", async () => {
+          expect(registerVehicleSpy).toHaveBeenCalledWith(
+            expect.objectContaining({
+              model: mockVehicle.model,
+              price: mockVehicle.price,
+              year: mockVehicle.year,
+              brandId: mockVehicle.brand.id,
+            })
+          );
+        });
+
+        it("should redirect me to vehicle list page after register", async () => {
+          expect(testLocation.pathname).toBe("/");
+        });
+      });
+
+      describe("But the response is rejected", () => {
+        beforeAll(() => {
+          registerVehicleSpy.mockRejectedValue({
+            data: "Marca não existe",
+            status: 404,
+          });
+        });
+
+        it("should show the error message", async () => {
+          const errorMsg = await screen.findByText(/Marca não existe/i);
+          expect(errorMsg).toBeInTheDocument();
+        });
+      });
+    });
+  });
+
+  describe("Update existing vehicle", () => {
+    beforeEach(async () => {
+      getAllBrandsSpy.mockResolvedValue(brands);
+      getVehicleSpy.mockResolvedValue({ data: mockVehicle });
+      await act(async () => setup(mockVehicle.id));
+    });
+
+    describe("It loads the vehicle data into the form", () => {
+      it("should render the vehicle model fetched from the param id", () => {
+        const input = screen.getByRole("textbox", { name: /modelo/i });
+        expect(input.value).toStrictEqual(mockVehicle.model);
+      });
+      it("should render the vehicle year fetched from the param id", () => {
+        const input = screen.getByRole("spinbutton", { name: /ano/i });
+        expect(input.value).toStrictEqual(mockVehicle.year);
+      });
+      it("should render the vehicle price fetched from the param id", () => {
+        const input = screen.getByRole("spinbutton", { name: /valor/i });
+        expect(input.value).toStrictEqual(mockVehicle.price);
+      });
+      it("should render the vehicle brand fetched from the param id", async () => {
+        const brand = screen.getByRole("combobox", { name: /marca/i });
+        expect(brand.value).toStrictEqual(brand[1].id);
+      });
+    });
+
+    describe("When submitting vehicle update form", () => {
+      beforeEach(async () => {
+        const model = screen.getByRole("textbox", { name: /modelo/i });
+        const brand = screen.getByRole("combobox", { name: /marca/i });
+
+        model.setSelectionRange(0, 4);
+        userEvent.type(model, "{backspace}Onix");
```

Comment:
Tem uma forma um pouco mais adequada de limpar o input: https://github.com/testing-library/user-event#clearelement

```suggestion
        userEvent.clear(model);
        userEvent.type(model, "Onix");
```